### ISSUE TITLE
Download Symbols from NAV2018 doesn't work.

### ISSUE THREAD

**Issue by robertoscaringella** [Microsoft/AL#2760]
Hi,
I installed NAV2018CU5 on our Company Server called nav2018.red.local.
I configurated the Cronus standard installation. On the NAV service I set to TRUE this fields:
- Enable Developer Service Endpoint
- Enable loading application symbol references at server startup
and then I changed NetFx40_LegacySecurityPolicy enable='false'
I installed the vsix in VS Code from the DVD of NAV2018CU6 because the previous version had a bugfix.
The Windows client and the Web Client run FINE from my laptop.
If I try to download symbols with this configuration from VS Code:

**launch.json**
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "al",
            "request": "launch",
            "schemaUpdateMode": "Recreate",
            "name": "TESTREPNAV18",
            "server": "http://nav2018.red.local/",
            "serverInstance": "DynamicsNAV110",
            "authentication": "Windows",
            "port": 7049
        }
    ]
}
**app.json**
{
  "id": "658e0dc0-6ce9-4415-b5a7-1d9a12318d1d",
  "name": "TestRep50100",
  "publisher": "Default publisher",
  "brief": "",
  "description": "",
  "showMyCode": false,
  "version": "1.0.0.2",
  "privacyStatement": "",
  "EULA": "",
  "help": "",
  "url": "",
  "logo": "",
  "capabilities": [],
  "dependencies": [],
  "screenshots": [],
  "platform": "11.0.0.0",
  "application": "11.0.0.0",
  "idRange": {
    "from": 50000,
    "to": 99999
  }
}
The download symbols and F5 (publish with debug) run FINE from my laptop!!!

Now, I recreate the same installation for an adding Database, a customer database. This NAV database come from an upgrade from NAV2013R2. I have upgraded the with the standard procedure raccomanded from MS: https://docs.microsoft.com/en-us/dynamics-nav/upgrading-the-data

I activated all the same fields, and the same setting. I changed the port setting.

**launch.json**
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "al",
            "request": "launch",
            "schemaUpdateMode": "Recreate",
            "name": "TESTREPNAV18",
            "server": "http://nav2018.red.local/",
            "serverInstance": "Customer2018",
            "authentication": "Windows",
            "port": 7349
        }
    ]
}
**app.json**
{
  "id": "658e0dc0-6ce9-4415-b5a7-1d9a12318d1d",
  "name": "TestRep50100",
  "publisher": "Default publisher",
  "brief": "",
  "description": "",
  "showMyCode": false,
  "version": "1.0.0.2",
  "privacyStatement": "",
  "EULA": "",
  "help": "",
  "url": "",
  "logo": "",
  "capabilities": [],
  "dependencies": [],
  "screenshots": [],
  "platform": "11.0.0.0",
  "application": "11.0.0.0",
  "idRange": {
    "from": 50000,
    "to": 99999
  }
}
The Windows client and the Web Client run FINE from my laptop.

The download symbols and F5 (publish with debug) run **DOESN'T WORK** from my laptop!!!
Output Error:

[2018-06-29 16:22:31.09] Using reference symbols cache path: ...\Progetti\CONSEA\Progetti AL\NAV18\TestRep50100\.alpackages
[2018-06-29 16:22:31.09] Sending request to http://nav2018.red.local:7349/Customer2018/dev/packages?publisher=Microsoft&appName=Application&versionText=11.0.0.0
[2018-06-29 16:22:31.09] Sending request to http://nav2018.red.local:7349/Customer2018/dev/packages?publisher=Microsoft&appName=System&versionText=11.0.0.0
[2018-06-29 16:22:38.53] The request for path /Customer2018/dev/packages?publisher=Microsoft&appName=System&versionText=11.0.0.0 failed with code NotFound. Reason: No published package matches the provided arguments.

Have you some suggestion? Could you help me to solve this problem?

Thank you in advanced,
Roberto.

**Comment by thpeder** [comment]
Hi @robertoscaringella,
I think you didn't publish the System symbols to your database. They are located on the DVD and you can see here how to publish them properly, [Upgrading the data](https://docs.microsoft.com/en-us/dynamics-nav/upgrading-the-data#AddExtensions)

**Comment by robertoscaringella** [comment]
Hi thpeder,
I follow the doc...
Step1:
Publish-NAVApp -ServerInstance <ServerInstanceName> -Path <SymbolFilePath> -PackageType SymbolsOnly

Step2:
finsql.exe Command=generatesymbolreference, Database=<MyDatabaseName>, ServerName=<DatabaseServerName>\<DatabaseInstance>

Step3:
The finsql.exe run for several minutes and use many CPU and memory, and when finished, I look on the client folder, but I don't see any log file (navcommandresult.txt or naverrorlog.txt).

Step4: I try to download again the symbols but doesn't work.
Maybe depend from the version of application. Because the link download is for: Microsoft_Application_11.0.19738.0.app and my NAV version is CU5 (21836). Have you other suggestion?

**Comment by robertoscaringella** [comment]
Hi thpeder,
Great suggestion. I kept the from the DVD follow files:
- System.app
- Test.app
and I kept the application.app downloaded from the Standard Cronus

and from PowerShell ISE executed:

Import-Module 'C:\Program Files\Microsoft Dynamics NAV\110\Service\NavAdminTool.ps1'
Publish-NAVApp -ServerInstance CUSTOMER2018 -packagetype SymbolsOnly -Path C:\NewFolder\System.app -SkipVerification
Publish-NAVApp -ServerInstance CUSTOMER2018 -packagetype SymbolsOnly -Path C:\NewFolder\Test.app -SkipVerification
#Publish-NAVApp -ServerInstance CUSTOMER2018 -packagetype SymbolsOnly -Path C:\NewFolder\Application.app -SkipVerification

and Now the dowload of the symbols from VS Code **WORK FINE!** but I have a PROBLEM with to publish the app.

In details, the VS Code return this Error: "dev/apps?tenant=default&SchemaUpdateMode=synchronize failed with code InternalServerError. Reason: Cannot install apps due to the state of the tenant: OperationalDataUpgradePending."

After I opened the WebClient of customer, and from Extension Management I saw the estention published but I cannot Install because the webclient return this error:

"A call to Microsoft.Dynamics.Nav.Runtime.Apps.NavAppALInstaller.ALInstallNavApp failed with this message: Cannot install apps due to the state of the tenant: OperationalDataUpgradePending."

I opened the PowerSheel ISE and I compared the configuration of the tenant:

ServerInstance                         : MicrosoftDynamicsNavServer$CUSTOMER2018
DatabaseName                           : CUSTOMER2018
DatabaseServer                         : NAV2018
DatabaseUserName                       : 
**State                                  : OperationalDataUpgradePending**
DetailedState                          : 
DeletionState                          : Not deleted
IsInExclusiveAccessMode                : False
**TenantDataVersion                      : Uninitialized**
Id                                     : default
AlternateId                            : {}
AllowAppDatabaseWrite                  : True
NasServicesEnabled                     : False
RunNasWithAdminRights                  : False
EncryptionProvider                     : LocalKeyFile
AzureKeyVaultSettings                  : 
DefaultCompany                         : 
DefaultTimeZone                        : UTC
ExchangeAuthenticationMetadataLocation : https://outlook.office365.com/
AadTenantId                            : common
DisplayName                            : 
UrlName                                : 
EnvironmentType                        : Production

------

ServerInstance                         : MicrosoftDynamicsNavServer$DynamicsNAV110
DatabaseName                           : Demo Database NAV (11-0)
DatabaseServer                         : NAV2018
DatabaseUserName                       : 
**State                                  : Operational**
DetailedState                          : 
DeletionState                          : Not deleted
IsInExclusiveAccessMode                : False
**TenantDataVersion                      : 11.0.21836.0**
Id                                     : default
AlternateId                            : {}
AllowAppDatabaseWrite                  : True
NasServicesEnabled                     : False
RunNasWithAdminRights                  : False
EncryptionProvider                     : LocalKeyFile
AzureKeyVaultSettings                  : 
DefaultCompany                         : 
DefaultTimeZone                        : UTC
ExchangeAuthenticationMetadataLocation : https://outlook.office365.com/
AadTenantId                            : common
DisplayName                            : 
UrlName                                : 
EnvironmentType                        : Production

Please, could you help me?

Thank you,
Roberto.

**Comment by thpeder** [comment]
Hi @robertoscaringella,
I do not have much knowledge about that area, but it seems like you haven't completed the upgrade process. 
I will recommend you to contact Customer Support.

**Comment by robertoscaringella** [comment]
Hi Thomas,
I found a solution, during the upgrade procedure it was happened something wrong. I tried from the c/side to make a syncronization (Tools/Sync. Schema for all Tables/Force...) and NAV show the same state "OperationalDataUpgradePending"...near below I saw the command "Data Upgrade" I choose "Show Progress" and with result "Not Started" and so, I choose "Start" and I Syncronized all the table and now run all. I can Publish and Install the app from client and VC Code, and I che start the app from VS Code in debug mode...

Thank you,
Roberto.
